utils.py

```
@@ -2,104 +2,66 @@
-        capped_delay = min(max_delay, backoff_delay)
-        actual_delay = random.uniform(min_delay, capped_delay)
-
-    # Bolt ⚡: Replaced an inefficient countdown loop with a single time.sleep() call.
-    # The previous implementation woke the CPU every second, leading to unnecessary
-    # context switching. This change reduces CPU usage and improves timer accuracy
-    # by sleeping for the entire duration in one go.
-    if actual_delay > 0:
-        logger.debug(f"Waiting {actual_delay:.2f}s before next operation.")
-        time.sleep(actual_delay)
+
+def parse_uk_datetime(dt_str: str, format: str = "%d/%m/%Y %H%M"):
```

Ruff (0.15.9)

[error] Function argument `format` is shadowing a Python builtin

(A002)